pykafka/balancedconsumer.py

```diff
@@ -460,22 +473,16 @@ def _path_from_partition(self, p):
     def _remove_partitions(self, partitions):
         """Remove partitions from the zookeeper registry for this consumer.
 
-        Also remove these partitions from the consumer's internal
-        partition registry.
-
         :param partitions: The partitions to remove.
         :type partitions: Iterable of :class:`pykafka.partition.Partition`
         """
         for p in partitions:
-            assert p in self._partitions
+            # TODO pass zk node version to make sure we still own this node
```

Comment:
I'm not sure - I mostly thought of this because I had to remove the `assert p in self._partitions` here, and wanted something similar in its place - so if we'd follow the original behaviour of the `assert`, then it would raise an exception here.

I think that's probably also the safest option: it would mean the rebalance would abort, and a new rebalance would probably be underway anyway (because anything that caused this loss of ownership would trigger the watches?), and the new rebalance would reset everything we think we know about which nodes we own? .... ugh, that was the longer version of "I'm not sure" :)